sklearnex/linear_model/incremental_linear.py

```diff
@@ -0,0 +1,177 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+from daal4py.sklearn._utils import daal_check_version
+
+if daal_check_version((2023, "P", 100)):
+    import numpy as np
+    from sklearn.utils import check_array, gen_batches
+
+    from daal4py.sklearn._utils import control_n_jobs, run_with_n_jobs
+    from onedal._device_offload import support_usm_ndarray
+    from onedal.linear_model import (
+        IncrementalLinearRegression as onedal_IncrementalLinearRegression,
+    )
+
+    @control_n_jobs
+    class IncrementalLinearRegression:
```

Comment:
What is the need for separate (from usual LinearRegression) class?

Reply:
to align with sklearn where PCA and IncrementalPCA are different classes